lib/Dashboard/MailWidget.php

```diff
@@ -95,4 +124,88 @@ public function load(): void {
 			$this->accountService->findByUserId($this->userId)
 		);
 	}
+
+	/**
+	 * Get widget-specific search filter
+	 * @return string
+	 */
+	abstract public function getSearchFilter(): string;
+
+	/**
+	 * @param string $userId
+	 * @return Message[]
+	 * @throws ClientException
+	 * @throws ServiceException
+	 */
+	protected function getEmails(string $userId): array {
+		$accounts = $this->accountService->findByUserId($userId);
+		$filter = $this->getSearchFilter();
+		$emails = [];
+
+		foreach ($accounts as $account) {
+			$mailboxes = $this->mailManager->getMailboxes($account);
+			/** @var Mailbox[] $mailboxes */
+			$mailboxes = array_filter($mailboxes, function (Mailbox $mailbox) {
+				return $mailbox->isInbox();
+			});
+			foreach ($mailboxes as $mailbox) {
+				$searchResult = $this->mailSearch->findMessages(
+					$account,
+					$mailbox,
+					$filter,
+					null,
+					10
+				);
+				$emails = array_merge($emails, $searchResult);
```

Comment:
array_merge in a loop is considered a performance anti-pattern. I think this needs to be restructured a bit so that array_merge is only run once after the nested loops